Dockerfile

```diff
@@ -6,25 +8,51 @@ RUN apt-get update && apt-get install -y  --no-install-recommends \
 
 WORKDIR /build
 
-# Cache dependencies
+# Target binary name (e.g., mojave-node, mojave-sequencer, mojave-prover)
+ARG TARGET_BIN
+RUN case "$TARGET_BIN" in \
+	mojave-node|mojave-sequencer|mojave-prover) ;; \
+	*) echo "Invalid TARGET_BIN=$TARGET_BIN"; exit 1 ;; \
+	esac
+
+# Optional build flags
+ARG BUILD_FLAGS=""
+
+# Cache deps
 COPY Cargo.toml Cargo.lock ./
 RUN mkdir src && echo "fn main() {}" > src/main.rs
-RUN cargo build --release || true
+RUN cargo build --release --bin ${TARGET_BIN} || true
 
+# Build
 COPY . .
-# Optional build flags
-ARG BUILD_FLAGS=""
-RUN cargo build --release $BUILD_FLAGS
+RUN cargo build --release --bin ${TARGET_BIN} ${BUILD_FLAGS}
 
-FROM debian:bookworm-slim
+# Runtime
+FROM debian:bookworm-slim AS runtime-base
+
+ARG TARGET_BIN
 
 RUN apt-get update && apt-get install -y --no-install-recommends \
 	libssl3 ca-certificates curl \
 	&& rm -rf /var/lib/apt/lists/*
 
-COPY data /usr/local/bin/data
-COPY --from=builder /build/target/release/mojave-node /usr/local/bin/mojave-node
-COPY --from=builder /build/target/release/mojave-prover /usr/local/bin/mojave-prover
-COPY --from=builder /build/target/release/mojave-sequencer /usr/local/bin/mojave-sequencer
-EXPOSE 8545
-ENTRYPOINT ["/usr/local/bin/mojave-sequencer"]
+COPY --from=builder /build/target/release/${TARGET_BIN} /usr/local/bin/${TARGET_BIN}
+
+RUN printf '%s\n' \
+	'#!/bin/sh' \
+	'set -e' \
+	'exec "${APP_BIN:-/usr/local/bin/'"${TARGET_BIN}"'}" "$@"' \
+	> /usr/local/bin/entrypoint.sh && chmod +x /usr/local/bin/entrypoint.sh
+ENTRYPOINT ["/usr/local/bin/entrypoint.sh"]
+
+FROM runtime-base AS mojave-node
+COPY data /data
+EXPOSE 8545 30304
+
+FROM runtime-base AS mojave-sequencer
+COPY data /data
+EXPOSE 1739
+
+FROM runtime-base AS mojave-prover
+
+FROM ${TARGET_BIN}
```

Comment:
we don't want to maintain this, and build all in one. either:
- create 3 separate dockerfiles
- keep it based on the `TARGET_BIN` pick, and let the responsibility to pick what target on other places